api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -228,21 +228,32 @@ private void ensureDrugOrderAutoExpirationDateIsSet(Order order) {
 			((DrugOrder) order).setAutoExpireDateBasedOnDuration();
 		}
 	}
-
+	
 	private void ensureOrderTypeIsSet(Order order, OrderContext orderContext) {
```

Comment:
@tendomart did you review the whole of this method?

Reply:
Let me do it again.

Reply:
@dkayiwa  any hints on someother way i may re-write the conditions for `List<OrderType>orderTypes = new ArrayList<>();`
?
Changing `if((orderTypes.size() > 1 || orderTypes == null) && order instanceof DrugOrder){}`

to `if((orderTypes.size() > 1 || orderTypes.size() == 0) && order instanceof DrugOrder){}

`and `if((orderTypes.size() > 1 || orderTypes.size() == "") && order instanceof DrugOrder){} doesnot work either`.

Reply:
@tendomart basing on your code, can `orderTypes` ever be null?

Reply:
No it cannot , it's actually fundamentally wrong ...But earlier changing to  `orderTypes == ""  `  got me into meaningless errors .

Reply:
@dkayiwa  how is it possible that `throw new AmbiguousOrderException("Order.cannot.have.more.than.one");` is striclty bound to the method , in that trying to throw any other exception for a new feature would not be caught ?

Reply:
Before even looking at that exception, the first step would be fixing your code so that it logically makes sense. Not so?

Reply:
It should run now

Reply:
> Before even looking at that exception, the first step would be fixing your code so that it logically makes sense. Not so?

You are right .